### PR TITLE
Collapsable sidebar on mobile

### PR DESCRIPTION
**Pull request by mudziok**
0 clue how to implement it closing on route change XD
i hope ive abstracted it in the correct place and really sorry if i didn't I have 0 confidence in my angular skills XD

### CODE REVIEW

**Comment by AdrianKokot** [comment]
> 0 clue how to implement it closing on route change XD

As router links are deep in the tree and will change over sidebar lifecycle, I think the only and reasonable way is to listen for router events and close sidebar on `NavigationStart` or `NavigationEnd` event.

You could do that in this way:

```ts
  constructor(readonly router: Router) {
    router.events
      .pipe(
        takeUntilDestroyed(),
        filter((e) => e instanceof NavigationEnd),
      )
      .subscribe(() => {
        this.open = false;
      });
  }
```
But this won't work until you change `ChangeDetectionStrategy` to `ChangeDetectionStrategy.Default`.

And the version working for `OnPush` strategy:

```ts
export class SidebarComponent implements AfterViewInit {
  @ViewChild(TuiActiveZoneDirective)
  private _button!: TuiActiveZoneDirective;

  open$!: Observable<boolean>; // *tuiSidebar="(open$ | async) ?? false"

  private readonly _router = inject(Router);

  ngAfterViewInit() {
    this.open$ = merge(
      this._button.tuiActiveZoneChange,
      fromEvent(this._button['el'].nativeElement, 'click').pipe(
        map(() => true)
      ),
      this._router.events.pipe(
        filter((e) => e instanceof NavigationEnd),
        map(() => false)
      )
    ).pipe(distinctUntilChanged());
  }
}
```

File: frontend/src/app/root.component.html
Comment (AdrianKokot) [inline frontend/src/app/root.component.html:1]: You can simplify this template with `*tuiLet`. `*tuiLet` is custom directive from taiga-ui that allows you to have variable in template from some expression and in this way you can skip multiple `| async` subscriptions. `*polymorpheousOutlet` is also taiga-ui thing and is some kind of extension of `*ngTemplateOutlet` so you can also skip that. For example:

```html
<tui-root *tuiLet='side.leftSide$ | async as leftSide'>
<!-- ... --->
    <reunice-sidebar
      menu
      *ngIf="leftSide !== null"
      class="sidebar-mobile"
    >
      <ng-container *polymorpheusOutlet="leftSide"></ng-container>
    </reunice-sidebar>
<!-- and same for sidebar-desktop and you can now remove ng-template #sidebar from the bottom--->
```

**Review by AdrianKokot — CHANGES_REQUESTED** [review]
Overall, looking good, just add the sidebar collapse on route change

File: frontend/src/app/root.component.html
Comment (mudziok) [reply frontend/src/app/root.component.html:1]: Thank you so so much with help on the collapse feature, ur the best!

Comment (mudziok) [reply frontend/src/app/root.component.html:1]: Also this let thingie seems really cool I'll make sure to read more docs on the directives tui provides

**Review by AdrianKokot — APPROVED** [review]

**Pull request merged**